src/extension.ts

```diff
@@ -94,10 +111,27 @@ export async function activate(context: vscode.ExtensionContext): Promise<vscode
 	return Promise.resolve(context);
 }
 
+export async function _stopExistingDfaRun(context: vscode.ExtensionContext, outputChannel: vscode.LogOutputChannel) {
+	const pid = context.workspaceState.get(Constants.GLOBAL_DFA_PROCESS);
+	if (pid) {
+		try {
+			process.kill(pid as number, SIGKILL);
+			await vscode.window.showInformationMessage(messages.graphEngine.dfaRunStopped);
+		} catch (e) {
+			const errMsg = e instanceof Error ? e.message : e as string;
+			outputChannel.appendLine('Failed killing DFA process.');
+			outputChannel.appendLine(errMsg);
+		}
+	} else {
+		await vscode.window.showInformationMessage(messages.graphEngine.noDfaRun);	
+	}
+	void context.workspaceState.update(Constants.GLOBAL_DFA_PROCESS, undefined);
```

Comment:
Question: Is there a reason you're `void`ing this promise instead of `await`ing it or just returning it?

Reply:
I wanted the function to proceed without waiting for the cache update. I also saw this pattern for these cache updates in other repositories.

Reply:
Fair enough. Please add a comment in the code explaining that, though.